python/ray/experimental/data/datasource/csv_datasource.py

```diff
@@ -0,0 +1,25 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import pyarrow
+
+from ray.experimental.data.datasource.file_based_datasource import (
+    FileBasedDatasource)
+
+
+class CSVDatasource(FileBasedDatasource):
+    """CSV datasource, for reading CSV files.
```

Comment:
```suggestion
    """CSV datasource, for reading and writing CSV files.
```

Reply:
I'll go ahead and add this since a broader discussion is out of scope for this PR, but I'm still of the opinion that we should either have `DataSource` (for reading) and `DataSink` (for writing) classes (or a similar split, e.g. `DataReader` and `DataWriter`), or we should rename it to something other than `DataSource` if it's going to house both reading and writing. A "data source" abstraction being used for writing isn't very intuitive to me, since we're not pulling data from a source, we're pushing data to a destination.

Reply:
Yeah I was debating that. Though it's not that unusual, and having a single class reduces the number of things floating around https://help.alteryx.com/current/designer/data-sources